specs/WinRT/MakeMSIX-spec.md

```diff
@@ -0,0 +1,959 @@
+    };
+
+    /// <summary>
+    /// Options for unpacking a package
+    /// </summary>
+    [contract(MakeMSIXContract, 1)]
+    runtimeclass UnpackOptions
+    {
+        /// <summary>
+        /// Created by callers to pass options into UnpackPackage
+        /// </summary>
+        UnpackOptions();
+
+        /// <summary>
+        /// The output folder to unpack the package into.
+        /// </summary>
+        String UnpackedPackageRootDirectory{ get; set; };
+        /// <summary>
+        /// If true, the operation will overwrite existing files in the UnpackedPackageRootDirectory
+        /// when unpacking.
+        /// If false, the operation will fail if a file already exists.
+        /// </summary>
+        Boolean OverwriteFiles{ get; set; };
+    };
+
+    /// <summary>
+    /// Options for unpacking a package
+    /// </summary>
+    [contract(MakeMSIXContract, 1)]
+    runtimeclass UnbundleOptions
+    {
+        /// <summary>
+        /// Created by callers to pass options into UnpackPackage
+        /// </summary>
+        UnbundleOptions();
+
+        /// <summary>
+        /// The output folder to unpack the bundle package into.
+        /// </summary>
+        String UnbundledPackageRootDirectory{ get; set; };
+        /// <summary>
+        /// If true, the operation will overwrite existing files in the UnbundledPackageRootDirectory
+        /// when unpacking.
+        /// If false, the operation will fail if a file already exists.
+        /// </summary>
+        Boolean OverwriteFiles{ get; set; };
+    };
+
+    /// <summary>
+    /// 
+    /// </summary>
+    [contract(MakeMSIXContract, 1)]
+    runtimeclass CreateMountableImageOptions
+    {
+        /// <summary>
+        /// Created by callers to pass options into CreateMountableImage
+        /// </summary>
+        CreateMountableImageOptions();
+
+        /// <summary>
+        /// Output path for the image file.
+        /// Valid file extensions are vhd, vhdx, and CIM.
+        /// </summary>
+        String ImageFilePath{ get; set; };
+
+        /// <summary>
+        /// Root directory path for packages inside the image.
+        /// Defaults to "WindowsApps"
+        /// </summary>
+        String PackageRootDirectoryInImage{ get; set; };
+
+        /// <summary>
+        /// If true, the operation will overwrite an existing file in the ImageFilePath
+        /// when unpacking.
+        /// If false, the operation will fail if the file already exists.
+        /// </summary>
+        Boolean OverwriteFiles{ get; set; };
+    };
+
+    /// <summary>
+    /// Static methods for creating and unpacking packages.
+    /// </summary>
+    [contract(MakeMSIXContract, 1)]
+    runtimeclass MakeMSIXManager
+    {
+        /// <summary>
+        /// Creates packages from the directoryPathToPack as specified by the packOptions.
+        /// </summary>
+        static Windows.Foundation.IAsyncAction Pack(String directoryPathToPack, PackOptions packOptions);
```

Comment:
Naming

`CreateKozaniPackage()` to create a Kozani package from an existing package? But `Pack()`, not `CreatePackage()`?

The usual WinRT API naming for APIs works better, even if terms differ from tool command line syntax e.g.

* pack = CreatePackage()
* bundle = CreateBundle()
* unpack = ExtractPackage()
* unbundle = ExtractBundle()
* ??? = CreateKozaniLocalVariant() or maybe just CreateKozani()
* ??? = CreateMountableImage()

P.S. .NET calls it ZipFile.CreateFromDirectory() and ZipFile.ExtractToDirectory() hence the Create+Extract verbs here.